ntbs-service/Pages/Migration/Index.cshtml.cs

```diff
@@ -33,10 +48,49 @@ public IActionResult OnGetAsync()
 
         public async Task<IActionResult> OnPostAsync()
         {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
             var requestId = HttpContext.TraceIdentifier;
 
-            Results = await _service.ImportNotificationsAsync(requestId, NotificationId);
+            if (UploadedFile != null)
+            {
+                var notificationIds = await GetIdListFromFile(UploadedFile);
+                var IdGroups = splitList(notificationIds, 10);
+
+                foreach (var idGroup in IdGroups)
+                {
+                    BackgroundJob.Enqueue<INotificationImportService>(x => x.ImportNotificationsByIdAsync(requestId, idGroup));
+                }
+            }
+            else if (CutoffNotificationDate != null)
+            {
+                CutoffNotificationDate.TryConvertToDateTimeRange(out DateTime? dateRangeStart, out DateTime? dateRangeEnd);
+                Results = await _service.ImportNotificationsByDateAsync(requestId, (DateTime) dateRangeStart);
+            }
+
             return Page();
         }
+
+        private async Task<List<string>> GetIdListFromFile(IFormFile file)
+        {
+            var ids = new List<string>();
+            using (var reader = new StreamReader(file.OpenReadStream()))
+            {
+                while (reader.Peek() >= 0)
+                    ids.Add(await reader.ReadLineAsync()); 
+            }
+            return ids;

```

Comment:
If this is reading in a notificationId in the sense of NTBS, this could probably have `int.TryParse` logic - if it is a legacyId (which I think needs to be a string to accomodate the distinct formats - then I would suggest that the ImportService could do with highlighting this

Reply:
yes, these are legacy ids. agree that we could make that distinction stronger throughout the feature!